analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -62,14 +63,14 @@ private enum DeclarationType
                         return;
                     }
 
-                    localDeclaration?.Declaration?.Variables
-                        .Where(v => v?.Identifier != null)
-                        .Select(v => new { Syntax = v, Symbol = c.SemanticModel.GetDeclaredSymbol(v) })
-                        .Where(ss => ss.Symbol != null)
-                        .Where(ss => IsInitializedWithCompatibleConstant(ss.Syntax, c.SemanticModel, declaredType))
-                        .Where(ss => !HasMutableUsagesInMethod(ss.Syntax, ss.Symbol, c.SemanticModel))
+                    localDeclaration.Declaration?.Variables
+                        .Where(v => v is { Identifier: { } }
+                            && c.SemanticModel.GetDeclaredSymbol(v) is { } symbol
```

Comment:
can you move `c.SemanticModel.GetDeclaredSymbol(v)` inside `HasMutableUsagesInMethod` (the only place it's used) and lazy load it only if necessary (`MatchesIdentifier`)?